src/test/java/org/mule/extension/aggregator/CommonAggregatorsTestCase.java

```diff
@@ -93,6 +93,30 @@ public void scheduledAggregationNotExecutedAfterSize() throws Exception {
     assertRouteExecutedNTimes(LISTENER_ROUTE_KEY, 1);
   }
 
+  @Test
+  @Description("scheduled period aggregation is not executed after size aggregation even if another is scheduled")
+  public void scheduledAggregationNotExecutedAfterSizeWithAnotherAggregationScheduled() throws Exception {
+    final String flowName = "scheduledAggregationNotExecuted";
+    final String payload = "lrm";
+    flowRunner(flowName).withPayload(payload).run();
+    //Wait until task is scheduled
+    waitForAggregatorTask(0);
+
+    flowRunner(flowName).withPayload(payload).run(); //Add one more element
+    flowRunner(flowName).withPayload(payload).run(); //Should trigger aggregation
+
+    flowRunner(flowName).withPayload(payload).run(); //Should schedule a new one
+
+    sleep(200); //After sleep, first scheduled task should have been released and second one should still need some time for registering
+
+    assertRouteExecutedNTimes(AGGREGATION_COMPLETE_ROUTE_KEY, 1);
+    assertRouteExecutedNTimes(LISTENER_ROUTE_KEY, 1); //Nothing should be executed before the 200ms mark
+
+    sleep(100); //Second task should be executed
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "Thread.sleep()". ([squid:S2925](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2925))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-core-extensions-mule-aggregators-module&pullRequest=43&issues=AWld7Rn3otIXnmsURTye&open=AWld7Rn3otIXnmsURTye)

[ISSUE_KEY]: AWld7Rn3otIXnmsURTye